### DISCUSSION TITLE
Handle connection

### DISCUSSION THREAD

**Discussion by AlexNatrix** [xyflow/xyflow#3547] [Q&A] [answered]
Hi guys.

I wanna to start connection, not from handle-mouse interaction, but from Node submenu (by clicking button in that submenu). I have node, i have submenu, i can get handle, but what state/event should i start/enable for that handle. Im trying to search in sources and here handlePointerDown , but i think its for internal purpose.

**Comment by bcakmakoglu** [comment, marked as answer]
This would be sort of tricky because what should the edge use as it's source handle when the submenu is closed (assuming the menu can be opened and closed)?

Technically you could make menu items a handle by just wrapping them with the handle component and overwriting the default handle styles.
When a connection is done, you could overwrite the source handle id to be that of a handle that might be on your node at all times and not just available in a menu.
